src/core/build_target.go

```diff
@@ -1430,6 +1430,11 @@ func (target *BuildTarget) AddNamedOutput(name, output string) {
 	target.namedOutputs[name] = target.insert(target.namedOutputs[name], output)
 }
 
+// AddOutput adds a new output to the target if it's not already there.
+func (target *BuildTarget) ShouldSandbox() bool {
```

Comment:
What's the point of this function? It could be useful to hide Sandbox but I'm not sure we need to - also TestSandbox is separate which this doesn't account for.

Reply:
There's a target interface that the process stuff uses that looks at a subset of the functions on target. I can't expose fields through an interface so needed to add this. I guess we might have this be `ShouldSandboxNet()` and `ShouldSandboxMount()` and have `target.Sandbox` be `all`, `none`, `net`, or `mount`.

Reply:
I don't think this is quite right. We need to check both Sandbox and TestSandbox. Will have to add another param to the exec then. :(